stdlib/assert.mt

```diff
@@ -174,43 +174,43 @@ defmodule Assert
 
     #doc <(other) -> self
     #| Operator alias for `less_than(other)`.
-    def <(other)
+    def <(other) : Assertion
       less_than(other)
     end
 
     #doc <=(other) -> self
     #| Operator alias for `less_or_equal(other)`.
-    def <=(other)
+    def <=(other) : Assertion
       less_or_equal(other)
     end
 
     #doc ==(other) -> self
     #| Operator alias for `equals(other)`.
-    def ==(other)
+    def ==(other) : Assertion
       equals(other)
     end
 
     #doc <=(other) -> self
     #| Operator alias for `does_not_equal(other)`.
-    def !=(other)
+    def !=(other) : Assertion
       does_not_equal(other)
     end
 
     #doc >=(other) -> self
     #| Operator alias for `greater_or_equal(other)`.
-    def >=(other)
+    def >=(other) : Assertion
       greater_or_equal(other)
     end
 
     #doc >(other) -> self
     #| Operator alias for `greater_than(other)`.
-    def >(other)
+    def >(other) : Assertion
       greater_than(other)
     end
 
     #doc is_a(other : Type) -> self
     #| Assert that the value is an instance of `type`.
-    def is_a(other : Type)
+    def is_a(other : Type) : Assertion
```

Comment:
I think this should be `Object` now, when I assert a given type is an object this will blow up.
Example:
```
test = %Type{}
assert(test).is_a(Object)
```

Reply:
Now, I hope Im not missing something obvious here, but.

I don't see why `Assertion#is_a` should return an `Object` if thats what you mean :eyes:
In fact I don't see that the return type of `is_a`, at all is relevant to the code sample you gave.
Pardon my ignorance, if I am so.

Reply:
That's an interesting point. I'm not actually sure what would happen there. `Type` and `Object` have some interesting circularity in them, where `Object` re-uses methods defined on `Type` so that it can act as a `Type` without directly inheriting from it.

Because it's not actually inheriting, I think this code would fail as you say it would. However, `Object` itself should probably be considered a `Type`, since it is created as one (the equivalent of a `deftype` but done in the native kernel). One solution there might be to have a special case of `is_a` that matches `Object` specifically:

```ruby
def is_a(other : Object); self; end
def is_a(type : Type) : Assertion
  ...
end
```

Since everything in Myst is an `Object` by default, I think that just returning `self` is a safe behavior. The assertion itself wouldn't really mean anything in a real example.

The method would still return an `Assertion` object to keep the fluid interface, it would be the parameter restriction that would change to `Object`.

Reply:
I've gone ahead and implemented the two-clause work-around for now with a TODO note about changing the ancestry of `Object`.